routing/payment_lifecycle.go

```diff
@@ -226,11 +226,16 @@ func (p *paymentLifecycle) resumePayment() ([32]byte, *route.Route, error) {
 			log.Warnf("Failed to find route for payment %x: %v",
 				p.paymentHash, err)
 
+			routeErr, ok := err.(noRouteError)
```

Comment:
Can't this be observed via `FailureReason() == FailureReasonError`?

Reply:
`FailureReason` is only defined on  `noRouteError`